docs/csharp/nullable-references.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Nullable reference types
+description: This article provides an overview of nullable reference types, added in C# 8. You'll learn how the feature provides safety against null reference exceptions, for new and existing projects.
+ms.date: 12/03/2018
+---
+# Nullable reference types
+
+C# 8 introduces **nullable reference types** and **non-nullable reference types** that enable you to make important statements about the properties for reference type variables:
+
+- **A reference is not supposed to be null**. When variables aren't supposed to be null, the compiler enforces these rules: The variable must be initialized to a non-null value. The variable can never be assigned the value `null`. Because of those two rules, it's safe to de-reference the variable without first checking that it isn't null.
+- **A reference may be null**. When variables may be null, the compiler enforces different rules. The variable may only be de-referenced when the compiler can guarantee that the value isn't null. These variables may be initialized with the default `null` value, and may be assigned the value `null` in other code.
+
+This new feature provides significant benefits over the earlier reference variables where the design intent couldn't be determined from the variable declaration. The compiler didn't provide safety for reference types:
```

Comment:
I'm not sure that "the earlier reference variables" is clear -- reference variables haven't changed, but language features that handle them has. Perhaps "over the handling of reference variables in earlier versions of C#"?

Reply:
Great suggestion. Thanks @rpetrusha